app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/auth/GroupEnrolmentExtractor.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eoricommoncomponent.frontend.controllers.auth
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.eoricommoncomponent.frontend.domain.{EnrolmentResponse, GroupId}
+import uk.gov.hmrc.eoricommoncomponent.frontend.models.Service
+import uk.gov.hmrc.eoricommoncomponent.frontend.services.subscription.EnrolmentStoreProxyService
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class GroupEnrolmentExtractor @Inject() (enrolmentStoreProxyService: EnrolmentStoreProxyService) {
+
+  def hasGroupIdEnrolmentTo(groupId: String, service: Service)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
```

Comment:
This `ec` can be moved to the class. In case that more methods will use it, it can be defined in one place only.

But I'm wondering if it's necessary. It's only invoking the method below which doesn't have ec there.

EC Can be necessary for the Future, but you can try to get rid of this here or move it to class level